signer/tests/integration/transaction_coordinator.rs

```diff
@@ -263,198 +260,21 @@ fn mock_deploy_all_contracts(
-        client
-            .expect_estimate_fees()
-            .returning(|_, _, _| Box::pin(async { Ok(100) }));
-
-        client.expect_get_account().returning(move |_| {
-            Box::pin(async move {
-                Ok(AccountInfo {
-                    balance: 1_000_000,
-                    locked: 0,
-                    unlock_height: 0u64.into(),
-                    nonce,
-                })
-            })
-        });
-
-        client.expect_submit_tx().returning(move |tx| {
-            let tx = tx.clone();
-            let txid = tx.txid();
-            let broadcasted_transaction_tx = broadcasted_transaction_tx.clone();
-            Box::pin(async move {
-                broadcasted_transaction_tx
-                    .send(tx)
-                    .expect("Failed to send result");
-                Ok(SubmitTxResponse::Acceptance(txid))
-            })
-        });
-    })
-}
 
-fn mock_deploy_remaining_contracts_when_some_already_deployed(
-    nonce: u64,
-    broadcasted_transaction_tx: Sender<StacksTransaction>,
-) -> Box<dyn FnOnce(&mut MockStacksInteract)> {
-    Box::new(move |client: &mut MockStacksInteract| {
-        // We expect the contract source to be fetched 5 times, once for
-        // each contract. The first two times, it will return an
-        // ContractSrcResponse, meaning that the contract was already
-        // deployed.
-        client
-            .expect_get_contract_source()
-            .times(2)
-            .returning(|_, _| {
-                Box::pin(async {
-                    Ok(ContractSrcResponse {
-                        source: String::new(),
-                        publish_height: 1,
-                        marf_proof: None,
-                    })
-                })
-            });
-        // The remaining 3 times, it will return an error, meaning that the
-        // contracts were not deployed.
-        client
-            .expect_get_contract_source()
-            .times(3)
-            .returning(|_, _| {
-                Box::pin(async {
-                    Err(Error::StacksNodeResponse(
-                        mock_reqwests_status_code_error(404).await,
-                    ))
-                })
-            });
-
-        // All the following functions, `estimate_fees`, `get_account` and
-        // `submit_tx` are only called for the 3 contracts to be deployed
-        // and once for the deposit tx
         client
             .expect_estimate_fees()
-            .times(3)
             .returning(|_, _, _| Box::pin(async { Ok(100) }));
 
-        client.expect_get_account().times(3).returning(move |_| {
+        client.expect_get_account().returning(move |_| {
             Box::pin(async move {
                 Ok(AccountInfo {
                     balance: 1_000_000,
                     locked: 0,
                     unlock_height: 0u64.into(),
-                    nonce,
+                    nonce: 16,
                 })
             })
         });
```

Comment:
nit: given we override those in `mock_stacks_core`, should we remove them from `mock_deploy_all_contracts`?